chia/full_node/coin_store.py

```diff
@@ -268,7 +268,8 @@ async def get_coin_states_by_puzzle_hashes(
         coins = set()
         puzzle_hashes_db = tuple([ph.hex() for ph in puzzle_hashes])
         cursor = await self.coin_record_db.execute(
-            f'SELECT * from coin_record WHERE puzzle_hash in ({"?," * (len(puzzle_hashes) - 1)}?) '
+            f"SELECT * from coin_record INDEXED BY coin_puzzle_hash "
+            f'WHERE puzzle_hash in ({"?," * (len(puzzle_hashes) - 1)}?) '
```

Comment:
explicitly using this index is also done here: https://github.com/Chia-Network/chia-blockchain/pull/9352/files#diff-c6d0245a206470efd973e323f09abc2682faebb1a5040059cdf25ae01096c312R206